mmv1/templates/terraform/flatten_property_method.go.tmpl

```diff
@@ -14,7 +14,7 @@
   limitations under the License. */ -}}
 {{- define "flattenPropertyMethod" }}
 {{- if $.WriteOnly }}
-{{- else if $.CustomFlatten }}
+{{- else if and $.CustomFlatten (not $.ShouldIgnoreCustomFlatten) }}
```

Comment:
Not necessarily a blocker for this PR but we should probably plan on making a TGC-only version of this template if we keep needing to add TGC-specific logic to it.